src/app/core/products/product-api.service.ts

```diff
@@ -26,12 +26,8 @@ export class ProductApiService extends DataApiService<Product> {
     }
 
     getProducts(): Observable<Product[]> {
-        return forkJoin([
-            this.coffeeApiService.getCoffees(),
-            this.merchandiseApiService.getMerchandise(),
-        ]).pipe(
-            map((products: Product[][]) => flatten(products)),
-        );
+        return zip(this.coffeeApiService.getCoffees(), this.merchandiseApiService.getMerchandise())
+            .pipe(map((products: Product[][]) => flatten(products)));
```

Comment:
i agree with you this is hard to understand. Let's revisit at a later point. I think we can one/a combination of rxjs operators that do a better job of conveying what is happening here